reana_client/utils.py

```diff
@@ -123,7 +123,11 @@ def _prepare_kwargs(reana_yaml):
     try:
         with open(filepath) as f:
             reana_yaml = yaml.load(f.read(), Loader=yaml.FullLoader)
-
+        try:
+            _validate_sections(reana_yaml)
```

Comment:
Instead of implementing our own validation, I think it's simpler and more scalable to enrich the [reana.yaml JSON Schema](https://github.com/reanahub/reana-client/blob/master/reana_client/schemas/reana_analysis_schema.json) and define a more strict structure. Then we can catch the errors and display them properly.
Check the JSON Schema docs to see how to restrict to certain properties and so on: https://json-schema.org/understanding-json-schema/reference/object.html#additional-properties

Related issues:
- https://github.com/reanahub/reana-client/issues/404
- https://github.com/reanahub/reana-client/issues/486
- https://github.com/reanahub/reana-client/issues/220